Tests/DatadogTests/Datadog/Mocks/CoreMocks.swift

```diff
@@ -1282,29 +1282,68 @@ class CoreLoggerMock: CoreLogger {
     var criticalLog: RecordedLog? { criticalLogs.last }
 }
 
+/// `Telemtry` recording received telemetry.
+class TelemetryMock: Telemetry, CustomStringConvertible {
+    private(set) var debugs: [String] = []
+    private(set) var errors: [(message: String, kind: String?, stack: String?)] = []
+    private(set) var description: String = "Telemetry logs:"
+
+    func debug(id: String, message: String) {
+        debugs.append(message)
+        description.append("\n- [debug] \(message)")
+    }
+
+    func error(id: String, message: String, kind: String?, stack: String?) {
+        errors.append((message: message, kind: kind, stack: stack))
+        description.append("\n - [error] \(message), kind: \(kind ?? "nil"), stack: \(stack ?? "nil")")
+    }
+}
+
 extension DD {
     /// Syntactic sugar for patching the `dd` bundle by replacing `logger`.
     ///
-    /// It returns the `logger` and old version of `dd`, so it can be used inline:
     /// ```
     /// let dd = DD.mockWith(logger: CoreLoggerMock())
     /// defer { dd.reset() }
     /// ```
-    static func mockWith<CL: CoreLogger>(logger: CL) -> DDMock<CL> {
+    static func mockWith<CL: CoreLogger>(logger: CL) -> DDMock<CL, TelemetryMock> {
         let mock = DDMock(
             oldLogger: DD.logger,
-            logger: logger
+            oldTelemetry: DD.telemetry,
+            logger: logger,
+            telemetry: TelemetryMock()
         )
         DD.logger = logger
         return mock
     }
+
+    /// Syntactic sugar for patching the `dd` bundle by replacing `telemetry`.
+    ///
+    /// ```
+    /// let dd = DD.mockWith(telemetry: TelemetryMock())
+    /// defer { dd.reset() }
+    /// ```
+    static func mockWith<TM: Telemetry>(telemetry: TM) -> DDMock<CoreLoggerMock, TM> {
```

Comment:
Not so happy with having two different factory methods:
* `static func mockWith<CL: CoreLogger>(logger: CL) -> DDMock<CL, TelemetryMock> {`
* `static func mockWith<TM: Telemetry>(telemetry: TM) -> DDMock<CoreLoggerMock, TM>`

but I'd rather wait for another thing that we may add to `DD` global before finding a better shape for it.